### ISSUE TITLE
Help with label placement in plot recipe

### ISSUE THREAD

**Issue by juliohm** [JuliaPlots/Plots.jl#2118]
Consider this example with `GeoStatsBase#master`:

```julia
using GeoStatsBase
using Plots; gr(format=:png)

d = RegularGridData{Float64}(Dict(:z => rand(100,100),
                                  :y => randn(100,100),
                                  :w => randn(100,100)))

histplot(d)
```
![Screenshot from 2019-08-01 16-37-58](https://user-images.githubusercontent.com/3345261/62322510-9973f000-b47b-11e9-871c-d5cb6a8341ce.png)

The x label is repeated incorrectly for all subplots. The expected behavior is to have a different x label for each subplot. The plot recipe implementation is here: https://github.com/juliohm/GeoStatsBase.jl/blob/master/src/plotrecipes/histplot.jl

Could you please help with the `xguide` attribute? Where should I place it?

**Comment by daschw** [comment]
Put `xguide --> var` inside the `@series` block.

**Comment by juliohm** [comment]
Yes, I did it in the first block, but the second block overrides it. I am still not up-to-date with the way hierarchical attributes work in Plots.jl. I did refactor the code, and it looks like this now:

![Screenshot from 2019-08-01 17-10-07](https://user-images.githubusercontent.com/3345261/62324496-11dcb000-b480-11e9-84de-2a06fc63e029.png)

Now I need to find a way to put the numbers at the top of the vertical lines indicating the value it corresponds to in the x axis.
